### PR TITLE
fix: do not run containers as root by default in Helm chart

### LINKED ISSUE #13869: Helm chart is configured to run as root without ability to override
Helm is [configured](https://github.com/apache/superset/blob/9d0bb3a3ba38c2289d77e87dfe572c57c51ef7c7/helm/superset/templates/deployment.yaml#L53) to run containers as root so that it can [install packages](https://github.com/apache/superset/blob/9d0bb3a3ba38c2289d77e87dfe572c57c51ef7c7/helm/superset/templates/_helpers.tpl#L52). This is against security best practices and to top it all off, the chart does not allow editing this setting, and neither can the bootstrap be edited, so it doesn't crash when running as non-root. This makes the helm chart not suitable for secure production environments.

To fix this, it would be enough to allow editing the `runAsUser` field and the bootstrap script from `values.yaml`. Although in my opinion bad practices like this should not be promoted by the upstream, so it would be preferable if non-root was the default option, with [documentation](https://superset.apache.org/docs/installation/running-on-kubernetes#running-on-kubernetes) reflecting this.

I may have time to fix this later this week, but my schedule is quite tight so if someone has time to do this, please post here so I don't start duplicating someone's work.

### PR DESCRIPTION
**Pull request by megakoresh**
### SUMMARY
This allows the user to control bootstrap behavior to avoid running containers with uid 0. Fixes #13869 

Some additions to documentation might be needed. Let me know if that's the case.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A - No change in behavior

### TEST PLAN
N/A - Helm chart does not currently have tests

### ADDITIONAL INFORMATION

- [X] Has associated issue: #13869 
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: helm/superset/values.yaml [hunk 1]
```diff
@@ -21,15 +21,19 @@
...
-additionalAptPackages: {}
-  # - nano
+# User ID directive
+# Note that this user must have appropriate permissions to run the bootstrap script
+runAsUser: 1000
+
+# Install additional packages and do any other bootstrap configuration in this script
+# For production clusters it's recommended to build own image with this step done in CI
+bootstrapScript: |
+  #!/bin/bash
```
Comment (craig-rueda) [inline helm/superset/values.yaml:31]: Did we want to comment all of these lines? Typo?

Comment (megakoresh) [reply helm/superset/values.yaml:31]: Yes. I switched default uid to 1000 which will cause these lines to fail if those lines are uncommented. The purpose is to show users that this is a place to perform such tasks, but also that they must make their environment less secure by setting runAsUser value to 0.

Comment (craig-rueda) [reply helm/superset/values.yaml:31]: Understood. Can you please put it back to 0 and enable that script? Maybe just add some verbage indicating that this isn't something one should do in production? There's several folks that rely on that thing currently (including me 😀).

Comment (megakoresh) [reply helm/superset/values.yaml:31]: Ok, this is now done.

**Review by craig-rueda — APPROVED** [review]

**Pull request merged**